Formatter: toString overloads are not recognized anymore

mathias-lang-sociomantic opened the pull request:
This regression was introduced in #112. While changing the type internally used by the formatter,
it affected which toString overload can be used.
We now correctly recognize both toString overloads and issue a 'deprecation' message when size_t is encountered.

mihails-strasuns-sociomantic commented:
Regression test case would be really appreciated ;)

mihails-strasuns-sociomantic commented on src/ocean/text/convert/Formatter.d:
```diff
@@ -385,7 +385,14 @@ private void handle (T) (T v, FormatInfo f, FormatterSink sf, ElemSink se)
+        v.toString((cstring e) { se(e, f); });
+    else static if (is(typeof(v.toString((size_t delegate(cstring)).init))))
+    {
+        pragma(msg, "Deprecation: Please change stringOf to accept ",
```
`stringOf` -> `toString`?

mathias-lang-sociomantic commented:
> Regression test case would be really appreciated ;)

The issue is that it'll print a message to stdout during unittests :(

mihails-strasuns-sociomantic commented:
Not if you add it as `something_test.d` module.

Geod24 committed:
Formatter: toString overloads are not recognized anymore

This regression was introduced in #112. While changing the type internally used by the formatter,
it affected which toString overload can be used.
We now correctly recognize both toString overloads and issue a 'deprecation' message when size_t is encountered.

Geod24 commented:
> Not if you add it as something_test.d module.

That's an excellent idea. Separating the tests from the Formatter could also allow us to solve the license issue (</noteToSelf>).
Updated.

mihails-strasuns-sociomantic commented:
Thanks!
